File: src/main/scala/br/com/fabioaraujo/calcula/produto/dto/Item.scala
package br.com.fabioaraujo.calcula.produto.dto

import javax.persistence._

import br.com.fabioaraujo.calcula.produto.dto.TipoUnidade.TipoUnidade

import scala.beans.BeanProperty

@Entity
class Item () extends Serializable{

  @Id  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @BeanProperty var id : Long = _
  @BeanProperty var nome : String = _
  @BeanProperty var custo : Double = _
  @BeanProperty var tipoUnidade : TipoUnidade = _


}

File: src/main/java/br/com/fabioaraujo/calcula/produto/dto/Item.java
package br.com.fabioaraujo.calcula.produto.dto;

import java.io.Serializable;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Item implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String nome;
	private Double custo;

	@Enumerated(EnumType.STRING) private TipoUnidade tipoUnidade;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public Double getCusto() {
		return custo;
	}

	public void setCusto(Double custo) {
		this.custo = custo;
	}

	public TipoUnidade getTipoUnidade() {
		return tipoUnidade;
	}

	public void setTipoUnidade(TipoUnidade tipoUnidade) {
		this.tipoUnidade = tipoUnidade;
	}



}
